lib/platform-stats.js

```diff
@@ -199,3 +316,119 @@ export const updateTopMeasurementParticipants = async (createPgClient) => {
     Sentry.captureException(err)
   }
 }
+
+/**
+ * @param {import('pg').Client} pgClient
+ */
+export const aggregateAndCleanupRecentData = async (pgClient) => {
+  try {
+    await pgClient.query('BEGIN')
+
+    // The ON CONFLICT theoretically should not happen, because it's two days old data,
+    // which should not be updated anymore. But it's a good practice to have it.
+    await pgClient.query(`
```

Comment:
Thoughts on this?

Reply:
Thoughts on the `ON CONFLICT`, or the whole query?

Reply:
Ok. Do you concur that it shouldn't happen? A second pair of eyes on that conclusion would be helpful, in case I didn't think about some case or edge case

Reply:
The question is can it happen? If it can, we need to prepare for it. A shouldn't happen isn't safe enough.

Can you think of a case in which it could happen? When would we update 2 days old data?

Reply:
In this case, I was using "should" as synonymous with "can", in the sense that I believe that it can't, but we can never be 100% certain with code, without mathematical, formal verification.

As I think more about it, I am thinking we should do nothing on conflict, and maybe Error or Warning, given that we don't expect it to happen

Reply:
I agree, in this case it's good to add a log "this shouldn't have happened" instead of conflict resolution logic

Reply:
> In this case, I was using "should" as synonymous with "can", in the sense that I believe that it can't, but we can never be 100% certain with code, without mathematical, formal verification.

If you believe but don't know then we should handle the case with a log line 👍

Reply:
Addressed in https://github.com/filecoin-station/spark-evaluate/pull/311/commits/e3845c9bd36cfddb719459fe72c3f2389b94ecda

I just removed the ON CONFLICT clause, so that it would throw an error, the DB transaction would rollback, and we get the log output of the catch block.